### PR TITLE
To Do List Interactions

### PR DESCRIPTION
**Pull request by albertkantwi**
Within this pull request, I have incorporate the following modifications:
- Add event listener to the checkbox.
- Implement function clear all complete.
- Update array items into local storage.

I kindly request for review

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  checkboxes.forEach((checkbox) => {
+    checkbox.addEventListener('change', () => {
+      const index = Number(checkbox.id.split('-')[0]);
+      const status = todoTasks[index - 1].completed;
+      todoTasks[index - 1].completed = !status;
+      saveToLocalStorage();
+    });
+  });
+  saveToLocalStorage();
+};
```
Comment (muneebulrehman) [inline src/index.js:32]: As per the project requirements, methods related to the status updates `(completed: true/false)` should be in a new javascript file that should be imported as a module. Kindly move the code related to this functionality to a new file and then import that as a module. ⬇️
image

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  const completedIndexes = todoTasks.reduce((indexes, task, index) => {
+    if (task.completed) indexes.push(index);
+    return indexes;
+  }, []);
+
+  if (completedIndexes.length > 0) {
+    deleteTask(completedIndexes);
+    buildTask();
+  }
+});
```
Comment (muneebulrehman) [inline src/index.js:48]: As per the project requirements, to clear all completed tasks, you should be using the `filter()` method to delete all the completed tasks. This approach will be much simpler than the current approach you are using.
image

**Review by muneebulrehman — CHANGES_REQUESTED** [review]
Hi @albertkantwi , 

Good job so far! I see you put a lot of effort into this project. However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- No linter errors
- Descriptive PR
- All functionality working. 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi @albertkantwi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed changes requested


### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**